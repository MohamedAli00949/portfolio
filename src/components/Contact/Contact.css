/* start contact section */

section#Contact h2 {
    text-align: center;
    font-size: 40px;
    margin-block: 20px;
    color: #09c6f9;
    text-shadow: 2px 1px 3px #000000cf;
  }
  
  .inputs {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
  }
  
  .success {
    margin-block: 10px;
    background: green;
    padding: 20px;
    text-align: center;
    font-size: 20px;
    border-radius: 10px;
    box-shadow: 0 0 5px 0px black;
    color: white;
    width: fit-content;
    position: fixed;
    top: 20%;
    margin: 10px;
    right: 5px;
    z-index: 100;
    box-shadow: 0 0 5px 0px #e7e7e7;
    border: 2px solid #86ff86;
  }
  
  /* end contact section */