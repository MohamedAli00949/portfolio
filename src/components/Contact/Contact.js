import React, { lazy, useEffect, useState } from "react";
import { IoIosSend } from "react-icons/io";

const Button = lazy(() => import("@mui/material/Button"));
const Container = lazy(() => import("@mui/material/Container"));
const Input = lazy(() => import("../../utils/Inputs"));

const Contact = () => {
  const [successSend, setSuccessSend] = useState(false);
  const [messageData, setMessageData] = useState({
    name: "",
    email: "",
    phone: "",
    message: "",
  });

  useEffect(() => {
    if (successSend) {
      setTimeout(() => {
        setSuccessSend((pS) => !pS);
      }, 5000);
    }
  }, [successSend]);

  const handleSubmit = (e) => {
    e.preventDefault();

    setSuccessSend(true);
    console.log(messageData);
    setMessageData({ name: "", email: "", phone: "", message: "" });
  };

  return (
    <section
      id="Contact"
      style={{
        padding: "132px 0px",
        background: "#def8ff",
      }}
    >
      <Container fixed>
        <div>
          <h2>Contact</h2>
          <div>
            <div>
              <form className="inputs" onSubmit={handleSubmit}>
                {successSend ? (
                  <div className="success">
                    Thanks for sending the message.
                    <div>
                      <span className="theUpper"></span>
                      <span className="the"></span>
                    </div>
                  </div>
                ) : null}
                <Input
                  type="text"
                  name="Name"
                  placeholder="Enter your name"
                  required
                  value={messageData.name}
                  onChange={(e) => {
                    setMessageData({ ...messageData, name: e.target.value });
                  }}
                />
                <Input
                  type="email"
                  name="Email"
                  placeholder="Enter your email"
                  required
                  value={messageData.email}
                  onChange={(e) => {
                    setMessageData({ ...messageData, email: e.target.value });
                  }}
                />
                <Input
                  type="tel"
                  name="Phone"
                  placeholder="Enter your phone"
                  required
                  value={messageData.phone}
                  onChange={(e) => {
                    setMessageData({ ...messageData, phone: e.target.value });
                  }}
                />
                <Input
                  type="textarea"
                  name="Message"
                  placeholder="Write your message"
                  required
                  value={messageData.message}
                  onChange={(e) => {
                    setMessageData({
                      ...messageData,
                      message: e.target.value,
                    });
                  }}
                />
                <Button
                  aria-label="submit data"
                  variant="contained"
                  type="submit"
                  disabled={
                    messageData.name === "" ||
                    messageData.email === "" ||
                    messageData.phone === "" ||
                    messageData.message === ""
                  }
                  endIcon={<IoIosSend />}
                  style={{ textTransform: "capitalize" }}
                >
                  Submit the message
                </Button>
              </form>
            </div>
          </div>
        </div>
      </Container>
    </section>
  );
};

export default Contact;
