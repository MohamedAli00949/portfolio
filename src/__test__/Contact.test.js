import React, { Suspense } from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import { act } from "react-dom/test-utils";

import Contact from "../components/Contact/Contact";

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    container.setAttribute("id", "containeer")
    document.body.appendChild(container);
});

afterEach(() => {
    ReactDOM.unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("render a contact section test", async () => {
    await act(async () => {
        await render(<Suspense><Contact /></Suspense>);
    });

    container = await document.body;

    expect(container.querySelector("section").getAttribute("id")).toBe("Contact");

    expect(container.querySelector("section").style.background).toBe("rgb(222, 248, 255)");
    expect(container.querySelector("section").style.padding).toBe("132px 0px");

    expect(container.querySelector("h2").textContent).toBe("Contact");

    expect(container.querySelectorAll("input")[0].getAttribute("id")).toBe("Name");
    expect(container.querySelectorAll("input")[1].getAttribute("id")).toBe("Email");
    expect(container.querySelectorAll("input")[2].getAttribute("id")).toBe("Phone");
    expect(container.querySelector("textarea").getAttribute("id")).toBe("Message");

    expect(container.querySelector("button").textContent).toBe("Submit the message");

    expect(container).toMatchSnapshot();
})