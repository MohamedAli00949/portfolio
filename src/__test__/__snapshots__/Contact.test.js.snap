// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a contact section test 1`] = `
<body>
  <div
    id="containeer"
  />
  <div>
    <section
      id="Contact"
      style="padding: 132px 0px; background: rgb(222, 248, 255);"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
      >
        <div>
          <h2>
            Contact
          </h2>
          <div>
            <div>
              <form
                class="inputs"
              >
                <div
                  class="input-container"
                >
                  <input
                    autocomplete="off"
                    class="text-input"
                    id="Name"
                    name="Name"
                    placeholder="Enter your name"
                    required=""
                    type="text"
                    value=""
                  />
                  <label
                    class="label"
                    for="Name"
                  >
                    Name
                  </label>
                </div>
                <div
                  class="input-container"
                >
                  <input
                    autocomplete="off"
                    class="text-input"
                    id="Email"
                    name="Email"
                    placeholder="Enter your email"
                    required=""
                    type="email"
                    value=""
                  />
                  <label
                    class="label"
                    for="Email"
                  >
                    Email
                  </label>
                </div>
                <div
                  class="input-container"
                >
                  <input
                    autocomplete="off"
                    class="text-input"
                    id="Phone"
                    name="Phone"
                    placeholder="Enter your phone"
                    required=""
                    type="tel"
                    value=""
                  />
                  <label
                    class="label"
                    for="Phone"
                  >
                    Phone
                  </label>
                </div>
                <div
                  class="input-container"
                >
                  <textarea
                    autocomplete="off"
                    class="text-input"
                    id="Message"
                    name="Message"
                    placeholder="Write your message"
                    required=""
                    rows="5"
                    style="resize: none;"
                    type="textarea"
                  />
                  <label
                    class="label"
                    for="Message"
                  >
                    Message
                  </label>
                </div>
                <button
                  aria-label="submit data"
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
                  disabled=""
                  style="text-transform: capitalize;"
                  tabindex="-1"
                  type="submit"
                >
                  Submit the message
                  <span
                    class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M435.9 64.9l-367.1 160c-6.5 3.1-6.3 12.4.3 15.3l99.3 56.1c5.9 3.3 13.2 2.6 18.3-1.8l195.8-168.8c1.3-1.1 4.4-3.2 5.6-2 1.3 1.3-.7 4.3-1.8 5.6L216.9 320.1c-4.7 5.3-5.4 13.1-1.6 19.1l64.9 104.1c3.2 6.3 12.3 6.2 15.2-.2L447.2 76c3.3-7.2-4.2-14.5-11.3-11.1z"
                      />
                    </svg>
                  </span>
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</body>
`;
